tests/kafkatest/utils/security_config.py

```diff
@@ -0,0 +1,132 @@
+        """
+        ksPath = os.path.join(dir, 'test.keystore.jks')
+        ksPassword = 'test-ks-passwd'
+        keyPassword = 'test-key-passwd'
+        tsPath = os.path.join(dir, 'test.truststore.jks')
+        tsPassword = 'test-ts-passwd'
+        if os.path.exists(ksPath):
+            os.remove(ksPath)
+        if os.path.exists(tsPath):
+            os.remove(tsPath)
+        
+        Keytool.runcmd("keytool -genkeypair -alias test -keyalg RSA -keysize 2048 -keystore %s -storetype JKS -keypass %s -storepass %s -dname CN=systemtest" % (ksPath, keyPassword, ksPassword))
+        Keytool.runcmd("keytool -export -alias test -keystore %s -storepass %s -storetype JKS -rfc -file test.crt" % (ksPath, ksPassword))
+        Keytool.runcmd("keytool -import -alias test -file test.crt -keystore %s -storepass %s -storetype JKS -noprompt" % (tsPath, tsPassword))
+        os.remove('test.crt')
+
+        return {
+            'ssl.keystore.location' : ksPath,
+            'ssl.keystore.password' : ksPassword,
+            'ssl.key.password' : keyPassword,
+            'ssl.truststore.location' : tsPath,
+            'ssl.truststore.password' : tsPassword
+        }
+
+    @staticmethod
+    def runcmd(cmd):
+        proc = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+        proc.communicate()
+        if proc.returncode != 0:
+            raise subprocess.CalledProcessError(proc.returncode, cmd)
+
+
+class SecurityConfig:
+
+    PLAINTEXT = 'PLAINTEXT'
+    SSL = 'SSL'
+    SSL_DIR = "/mnt/ssl"
+    KEYSTORE_PATH = "/mnt/ssl/test.keystore.jks"
+    TRUSTSTORE_PATH = "/mnt/ssl/test.truststore.jks"
+
+    ssl_stores = Keytool.generate_keystore_truststore('.')
+
+    def __init__(self, node_account, security_protocol, template_props=""):
+        """
+        Initialize the security properties for the node and copy
+        keystore and truststore to the remote node if the transport protocol 
+        is SSL. If security_protocol is None, the protocol specified in the
+        template properties file is used. If no protocol is specified in the
+        template properties either, PLAINTEXT is used as default.
+        """
+
+        self.node_account = node_account
+        if security_protocol is None:
+            for line in template_props.split("\n"):
+                items = line.split("=")
+                if len(items) == 2 and items[0].strip() == 'security.protocol':
+                    security_protocol = str(items[1].strip())
+                    if security_protocol not in [SecurityConfig.PLAINTEXT, SecurityConfig.SSL]:
+                        raise Exception("Invalid security.protocol in template properties: " + security_protocol)
+        if security_protocol is None:
+            security_protocol = SecurityConfig.PLAINTEXT
+        if security_protocol == SecurityConfig.SSL:
+            node_account.ssh("mkdir -p %s" % SecurityConfig.SSL_DIR, allow_fail=False)
```

Comment:
This class seems a bit weird -- it's constructor seems to be used like a function in a bunch of places (taking the node account and putting files in place, where you don't save the `SecurityConfig` object, just create it to get the side-effects of the constructor), but with a few extra helpers added in that you only occasionally hold on to the object for.

Should we split out the constructor, which ensures we have all the settings, from the part that actually configures the node?

Originally I was thinking that using a mixin, this would work naturally -- the determination of the configs would happen in `__init__`, then `start` or `start_node` would setup the files on the host.